designs/2025-rule-tester-assertion-options/README.md

```diff
@@ -0,0 +1,391 @@
+                                                        message.suggestions === void 0,
+@@ -1157,6 +1174,10 @@ class RuleTester {
+                                                });
+
+                                                if (hasOwnProperty(error, "message")) {
++                                                       assert.ok(
++                                                               requireMessage !== "messageId",
++                                                               "The test run requires 'messageId' but the error (also) uses 'message'.",
++                                                       );
+                                                        assert.ok(
+                                                                !hasOwnProperty(error, "messageId"),
+                                                                "Error should not specify both 'message' and a 'messageId'.",
+@@ -1165,11 +1186,16 @@ class RuleTester {
+                                                                !hasOwnProperty(error, "data"),
+                                                                "Error should not specify both 'data' and 'message'.",
+                                                        );
++
+                                                        assertMessageMatches(
+                                                                message.message,
+                                                                error.message,
+                                                        );
+                                                } else if (hasOwnProperty(error, "messageId")) {
++                                                       assert.ok(
++                                                               requireMessage !== "message",
++                                                               "The test run requires 'message' but the error (also) uses 'messageId'.",
++                                                       );
+                                                        assert.ok(
+                                                                ruleHasMetaMessages,
+                                                                "Error can not use 'messageId' if rule under test doesn't define 'meta.messages'.",
+@@ -1239,22 +1265,22 @@ class RuleTester {
+                                                const actualLocation = {};
+                                                const expectedLocation = {};
+
+-                                               if (hasOwnProperty(error, "line")) {
++                                               if (hasOwnProperty(error, "line") || requireLocation) {
+                                                        actualLocation.line = message.line;
+                                                        expectedLocation.line = error.line;
+                                                }
```

Comment:
I think the error message should be more explicit in these cases, saying e.g., that `line` assertion is mandatory, rather than showing that `undefined` doesn't match the actual line.

Reply:
Changed